django/contrib/sitemaps/__init__.py

```diff
@@ -15,21 +15,22 @@ class SitemapNotFound(Exception):
     pass
 
 
-def ping_google(sitemap_url=None, ping_url=PING_URL):
+def ping_google(sitemap_url=None, ping_url=PING_URL, site_domain=None, is_secure=False):
     """
     Alert Google that the sitemap for the current site has been updated.
     If sitemap_url is provided, it should be an absolute path to the sitemap
     for this site -- e.g., '/sitemap.xml'. If sitemap_url is not provided, this
     function will attempt to deduce it by using urls.reverse().
     """
-    sitemap_full_url = _get_sitemap_full_url(sitemap_url)
+    sitemap_full_url = _get_sitemap_full_url(sitemap_url, site_domain, is_secure)
     params = urlencode({'sitemap': sitemap_full_url})
     urlopen('%s?%s' % (ping_url, params))
 
 
-def _get_sitemap_full_url(sitemap_url):
-    if not django_apps.is_installed('django.contrib.sites'):
-        raise ImproperlyConfigured("ping_google requires django.contrib.sites, which isn't installed.")
+def _get_sitemap_full_url(sitemap_url, site_domain=None, is_secure=False):
+    if not site_domain:
+        if not django_apps.is_installed('django.contrib.sites'):
```

Comment:
combine with above `if` - `if not site_domain and not django_apps.is_installed...`. save an indentation level.